src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,37 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add a movie to DB: " + movie);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Movie movie;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            movie = session.get(Movie.class, id);
+            session.close();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movie by id from DB: " + id);
+        }
+        return Optional.ofNullable(movie);
```

Comment:
Use try with finally, where you use transaction and try with resources with Read operations.
see https://mate-academy.github.io/jv-program-common-mistakes/hibernate/relations/relations_checklist